src/main/java/com/contentful/java/cda/interceptor/ContentfulUserAgentHeaderInterceptor.java

```diff
@@ -0,0 +1,299 @@
+       * @return a version of (1, 2, 3, dev)
+       * @throws NumberFormatException    if major, minor or patch are no numbers
+       * @throws IllegalArgumentException if numbers are negative.
+       * @throws IllegalArgumentException if stability cannot be read.
+       */
+      public static Version parse(String version) {
+        final Matcher matcher = VERSION_PATTERN.matcher(version);
+
+        if (matcher.find() && matcher.groupCount() == 4) {
+          // ignore first full matching result
+          final int major = Integer.parseInt(matcher.group(1));
+          final int minor = Integer.parseInt(matcher.group(2));
+          final int patch = Integer.parseInt(matcher.group(3));
+
+          if (major < 0) {
+            throw new IllegalArgumentException("Major version number cannot be negative!");
+          }
+          if (minor < 0) {
+            throw new IllegalArgumentException("Minor version number cannot be negative!");
+          }
+          if (patch < 0) {
+            throw new IllegalArgumentException("Patch version number cannot be negative!");
+          }
+
+          final String stability = parseStability(matcher.group(4));
+          return new Version(major, minor, patch, stability);
+        } else {
+          throw new IllegalArgumentException("Could not parse version: " + version);
+        }
+      }
+
+      private static String parseStability(String stability) {
+        // enforce uppercase
+        stability = stability.toUpperCase();
+
+        // check for hyphen
+        if (!stability.startsWith("-")) {
+          return null;
+        }
+        stability = stability.substring(1);
+
+        // check for consecutive letters
+        final Matcher matcher = STABILITY_PATTERN.matcher(stability);
+        if (matcher.find()) {
+          return matcher.group(0);
+        } else {
+          return null;
+        }
+      }
+
+      private final int major;
+      private final int minor;
+      private final int patch;
+      private final String stability;
+
+      /**
+       * Create a release version, omitting stability
+       *
+       * @param major How many breaking changes did this version release?
+       * @param minor How many additional backwards compatible changes were added?
+       * @param patch How many bugs were fixed in the release?
+       */
+      public Version(int major, int minor, int patch) {
+        this(major, minor, patch, null);
+      }
+
+      /**
+       * Create a version including a stability.
+       *
+       * @param major     How many breaking changes did this version release?
+       * @param minor     How many additional backwards compatible changes were added?
+       * @param patch     How many bugs were fixed in the release?
+       * @param stability Is this a stable version(null) or is this not on production?
+       */
+      public Version(int major, int minor, int patch, String stability) {
```

Comment:
I think the documentation about the parameters for this constructor is a bit misleading. I think we can omit "how many breaking changes did this version release" as really a version comes after a previous version, and you bump the major version number _if_ there were a breaking changes (it's binary not quantitative). Anyway, short story is that I think just giving an example here would suffice more than explaining each of the major, minor, patch with words.